mlir/lib/Dialect/ArmSME/Transforms/VectorLegalization.cpp

```diff
@@ -338,6 +338,105 @@ struct LegalizeTransferWriteOpsByDecomposition
   }
 };
 
+// Shuffles arith extend ops after vector.extract op.
```

Comment:
[nit] I find this form particularly easy to parse 😅 
```suggestion
//  arith.extend(vector.extract) -> vector.extract(arith.extend) 
```